activesupport/test/broadcast_logger_test.rb

```diff
@@ -307,6 +307,72 @@ def info(msg, &block)
       assert_equal true, @logger.error("Hello")
     end
 
+    test "with_level invokes its block one time, even when broadcasting to 2 or more loggers" do
+      invocations = 0
+      log1 = ::Logger.new(nil)
+      log2 = ::Logger.new(nil)
+
+      logger = BroadcastLogger.new(log1, log2)
+
+      invocations = 0
+      logger.with_level(:info) do
+        invocations += 1
+      end
+
+      assert_equal 1, invocations
+    end
+
+    test "with_level invokes its block one time, even when broadcasting to zero loggers" do
+      logger = BroadcastLogger.new()
+
+      invocations = 0
+      logger.with_level(:info) do
+        invocations += 1
+      end
+
+      assert_equal 1, invocations
```

Comment:
On `main` this test fails with:

```
NoMethodError: undefined method `with_level' for #<ActiveSupport::BroadcastLogger:0x0000000108200720 @broadcasts=[], @progname="Broadcast">
```

Because there are no internal broadcasts, and `respond_to_missing?` checks whether at least _one_ logger responds to the method, `BroadcastLogger` stops having a `with_logger` level when you take away its last logger, which doesn't seem like a good API.